bear.py

```diff
@@ -0,0 +1,176 @@
+
+    def eat_honey(self, field):
+        cell = field.get_cell_at_location(self.cell_row, self.cell_col)
+
+        if not cell.has_honey:
+            return
+
+        do_eat = random.random()
+
+        if do_eat <= self.smell:
+            print >> out_file, 'Bear {} ate honey'.format(self.name)
+            self.life_count += 1
+            self.activity_level /= 3
+            cell.remove_honey()
+
+    def do_you_defeat(self):
+        do_defeat = random.random()
+        return do_defeat <= self.cowardice
+
+    def set_cell(self, cell_row, cell_col):
+        self.cell_row = cell_row
+        self.cell_col = cell_col
+
+    def go_to_random_cell(self, field):
+        field.remove_bear(self.cell_row, self.cell_col)
+        random_empty_cell = field.get_random_cell_with_no_bear()
+        self.set_cell(random_empty_cell.row, random_empty_cell.col)
+        field.set_bear(self, self.cell_row, self.cell_col)
+
+    def die(self, field):
+        print >> out_file, 'Bear B{} dies'.format(self.name)
+        cell = field.get_cell_at_location(self.cell_row, self.cell_col)
+        cell.remove_bear()
+
+    def fight(self, another_bear, field):
+        print >> out_file, "F I G H T"
+        do_fight = random.random()
+        do_defeat = random.random()
+        if do_fight <= self.aggression:
+            print >> out_file, "Fight according to aggression level"
+            if do_defeat <= self.cowardice and not another_bear.do_you_defeat():
+                # I defeat
+                self.life_count = self.life_count * 0.9
+                another_bear.life_count = another_bear.life_count + self.life_count * 0.1
+                # I go to random cell
+                self.go_to_random_cell(field)
+                print >> out_file, 'B{} defeats, goes to r{} c{} '.format(self.name, self.cell_row, self.cell_col)
+            else:
+                # No one defeats - fight
+                my_fighting_coef = generate_fighting_coef()
+                his_fighting_coef = generate_fighting_coef()
+
+                while my_fighting_coef == his_fighting_coef:
+                    my_fighting_coef = generate_fighting_coef()
+                    his_fighting_coef = generate_fighting_coef()
+
+                # I win
+                if my_fighting_coef > his_fighting_coef:
+                    end_of_fight(field, self, another_bear, my_fighting_coef, his_fighting_coef)
+                    print >> out_file, 'B{} wins '.format(self.name)
+                # I loose
+                if his_fighting_coef > my_fighting_coef:
+                    end_of_fight(field, another_bear, self, his_fighting_coef, my_fighting_coef)
+                    print >> out_file, 'B{} looses '.format(self.name)
+        else:
+            # Aggression level doesn't let fighting
+            print >> out_file, "Agression not enough for fighting, go to random cell"
+            self.go_to_random_cell(field)
+
+        field.draw_field()
```

Comment:
Why is the fight function responsible for drawing the field and bears

Reply:
the printing is kind of debug information, it's either there or in move